src/Common/tests/System/Collections/IEnumerable.Generic.Tests.cs

```diff
@@ -57,19 +57,25 @@ public abstract partial class IEnumerable_Generic_Tests<T> : TestBase<T>
         /// If this property is set to true, the tests ensure that the exception is thrown. The default value is
         /// false.
         /// </summary>
-        protected virtual bool Enumerator_Current_UndefinedOperation_Throws => false;
+        protected virtual bool Enumerator_Generic_Current_UndefinedOperation_Throws => false;
 
         /// <summary>
-        /// The behavior of MoveNext at the end of the enumerable after modification is undefined for the generic
-        /// IEnumerable. It may either throw an InvalidOperationException or do nothing.
+        /// When calling Current of the enumerator before the first MoveNext, after the end of the collection,
+        /// or after modification of the enumeration, the resulting behavior is undefined. Tests are included
+        /// to cover two behavioral scenarios:
+        ///   - Throwing an InvalidOperationException
+        ///   - Returning an undefined value.
+        /// 
+        /// If this property is set to true, the tests ensure that the exception is thrown. The default value is
+        /// false.
         /// </summary>
-        protected bool MoveNextAtEndThrowsOnModifiedCollection => true;
+        protected virtual bool Enumerator_NonGeneric_Current_UndefinedOperation_Throws => true;
```

Comment:
Hmm. @ianhays, is this how you want things configured? We shouldn't duplicate every non generic test in the generic test classes.